llvm/lib/Target/RISCV/GISel/RISCVCallLowering.cpp

```diff
@@ -98,6 +98,36 @@ struct RISCVOutgoingValueHandler : public CallLowering::OutgoingValueHandler {
     MIB.addUse(PhysReg, RegState::Implicit);
   }
 
+  unsigned assignCustomValue(CallLowering::ArgInfo &Arg,
+                             ArrayRef<CCValAssign> VAs,
+                             std::function<void()> *Thunk) override {
+    const CCValAssign &VALo = VAs[0];
```

Comment:
Add an assert on VAs.size() being 2.

Reply:
The size isn't ==2. It is >=2. The caller has no idea how many entries this handler needs. It's continuously calling `slice` on an ArrayRef of all VAs to drop the ones that have been consumed. I'll add the assert for >=2